gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/filter/AbstractJWTFilter.java

```diff
@@ -342,7 +376,34 @@ protected boolean validateToken(HttpServletRequest request, HttpServletResponse
     return false;
   }
 
-  protected boolean verifyTokenSignature(final JWT token) {
+  protected boolean validateToken(final HttpServletRequest request,
+                                  final HttpServletResponse response,
+                                  final FilterChain chain,
+                                  final String tokenId)
+          throws IOException, ServletException {
+
+    if (tokenStateService != null) {
+      try {
+//        TokenMetadata metadata = tokenStateService.getTokenMetadata(tokenId);
```

Comment:
nit: can loose the comments here if you revisit this.

Reply:
I've created KNOX-2563 to provide the necessary metadata.